app/assets/javascripts/angular/controllers/dashboard_ctrl.js

```diff
@@ -74,8 +74,18 @@ angular.module("Prometheus.controllers")
     $scope.nextCycleRedraw();
   };
 
+  $window.history.replaceState({initial: true}, "Promdash - " + dashboardName, $window.location.href);
   $document.keydown(function(ev) {
-    if (ev.keyCode === 27) { // Escape keycode
+    var Z_KEY = 90;
+    var ESC_KEY = 27;
+    if (ev.keyCode === Z_KEY && document.activeElement.tagName !== "INPUT") {
```

Comment:
We should probably exempt textareas too. If we add them later (like if someone wants to have a larger expression input), we probably wouldn't notice during testing that pressing Z in a textarea zooms out, and only users would finally let us know.